app/models/call_session.rb

```diff
@@ -17,18 +17,28 @@ def authorized?
   end
 
   def save
-    Rails.cache.write(CallSession.session_key(@call_id),
-                      session_data,
-                      expires_in: EXPIRE_CALL_SESSION_IN)
+    CALL_SESSION_STORE_POOL.with do |connection|
+      RedisService
+        .new(connection)
+        .hmset_with_expiry(CallSession.session_key(@call_id),
+                           session_data,
+                           EXPIRE_CALL_SESSION_IN_SECONDS)
+    end
   end
 
   def kill
-    Rails.cache.delete(CallSession.session_key(@call_id))
+    CALL_SESSION_STORE_POOL.with do |connection|
+      connection.del(CallSession.session_key(@call_id))
```

Comment:
This can also be moved to `RedisService`

Reply:
Done.